src/useStorageValue/useStorageValue.ts

```diff
@@ -156,7 +156,9 @@ export function useStorageValue<
 
   const isFirstMount = useFirstMountState();
   const [state, setState] = useState<Type | null | undefined>(
-    options?.initializeWithValue && isFirstMount ? storageActions.current.fetch() : undefined
+    optionsRef.current?.initializeWithValue && isFirstMount
```

Comment:
There is no sence of using the ref value since it is only suitable during forst render, after re-render it wint take any effect -> no need to use ref.